sea-query after version v1.0.0-rc.6 does not compile

## Description
sea-query somewhere around v1.0.0-rc.6 does not compile
```
Checking sea-query v1.0.0-rc.11
error[E0080]: evaluation of constant value failed
   --> /Users/botafi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.11/src/value.rs:331:9
    |
331 |         panic!("the size of Value shouldn't be greater than 32 bytes")
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'the size of Value shouldn't be greater than 32 bytes', /Users/botafi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.11/src/value.rs:331:9
    |
note: inside `check_value_size`
   --> /Users/botafi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.11/src/value.rs:331:9
    |
331 |         panic!("the size of Value shouldn't be greater than 32 bytes")
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `VALUE_SIZE`
   --> /Users/botafi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.11/src/value.rs:327:31
    |
327 | pub const VALUE_SIZE: usize = check_value_size();
    |                               ^^^^^^^^^^^^^^^^^^
    = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not compile `sea-query` (lib) due to 1 previous error
```
I wanted to use the latest sea-orm as it has cloneable DB connection when using mock feature

## Steps to Reproduce

1. install sea-orm & sea-query latest RCs

### Expected Behavior

sea-query should compile

### Actual Behavior

It does not compile because of the above

### Reproduces How Often

always

## Versions
sea-orm = { version = "=2.0.0-rc.5", features = [
    "sqlx-postgres",
    "runtime-tokio-rustls",
    "macros",
    "postgres-array",
    "with-uuid",
    "debug-print",
    "mock"
] }
sea-query = "=1.0.0-rc.11"

cargo tree | grep sea-
│   ├── sea-orm v2.0.0-rc.5
│   │   ├── sea-orm-macros v2.0.0-rc.5 (proc-macro)
│   │   │   ├── sea-bae v0.2.1 (proc-macro)
│   │   ├── sea-query v1.0.0-rc.11
│   │   │   ├── sea-query-derive v1.0.0-rc.9 (proc-macro)
│   │   ├── sea-query-sqlx v0.8.0-rc.8
│   │   │   ├── sea-query v1.0.0-rc.11 (*)
│   ├── sea-query v1.0.0-rc.11 (*)
├── sea-orm v2.0.0-rc.5 (*)
├── sea-query v1.0.0-rc.11 (*)
├── sea-orm v2.0.0-rc.5 (*)
├── sea-query v1.0.0-rc.11 (*)
├── sea-orm v2.0.0-rc.5 (*)
├── sea-query v1.0.0-rc.11 (*)
├── sea-orm-migration v2.0.0-rc.5
│   ├── sea-orm v2.0.0-rc.5 (*)
│   ├── sea-orm-cli v2.0.0-rc.5
│   ├── sea-schema v0.17.0-rc.6
│   │   ├── sea-query v1.0.0-rc.11 (*)
│   │   ├── sea-query-sqlx v0.8.0-rc.8 (*)
│   │   ├── sea-schema-derive v0.3.0 (proc-macro)
├── sea-orm v2.0.0-rc.5 (*)
├── sea-query v1.0.0-rc.11 (*)
├── sea-orm v2.0.0-rc.5 (*)
├── sea-query v1.0.0-rc.11 (*)

rustc 1.85.0 (4d91de4e4 2025-02-17) stable-aarch64-apple-darwin (default)
even on recent nightly same issue

MacOS 15.4.1 (24E263)

--- a/src/value.rs
+++ b/src/value.rs
@@ -330,10 +330,25 @@ pub enum Value {
 /// Previously, the size was 24. We bumped it to 32 such that `String`
 /// can be unboxed.
 pub const VALUE_SIZE: usize = check_value_size();
+const MAX_VALUE_SIZE: usize = 32;
+
+/// When the `with-json` feature is enabled, the size of `Value` may exceed 32 bytes to 72 bytes if serde_json feature `preserve_order` is enabled as different Map implementation can be used.
+#[cfg(feature = "with-json")]
+const EXPECTED_VALUE_SIZE: usize = {
+    if size_of::<Option<Json>>() > MAX_VALUE_SIZE {
+        size_of::<Option<Json>>()
+    } else {
+        MAX_VALUE_SIZE
+    }
+};
+#[cfg(not(feature = "with-json"))]
+const EXPECTED_VALUE_SIZE: usize = MAX_VALUE_SIZE;
 
 const fn check_value_size() -> usize {
-    if std::mem::size_of::<Value>() > 32 {
-        panic!("the size of Value shouldn't be greater than 32 bytes")
+    if std::mem::size_of::<Value>() > EXPECTED_VALUE_SIZE {
+        panic!(
+            "the size of Value shouldn't be greater than the expected MAX_VALUE_SIZE (32 bytes by default)"
+        )
     }
     std::mem::size_of::<Value>()
 }

